### PR TITLE
Users & Reservations Endpoints, Removed BE Docker Files

### PR DESCRIPTION
**Pull request by AlexJoeb**
# Description

Implemented `/users` and `/reservations` routes. Created SwaggerUI API documentation for the created routes. Removed docker files.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Change Status

- [ ] Complete, tested, ready to review and merge
- [x] Complete, but not tested (may need new tests)
- [ ] Incomplete/work-in-progress, PR is for discussion/feedback

# Has This Been Tested?

- [ ] Yes
- - [ ] What was tested?
- [x] No
- - Unable to test without the Okta token.

# Checklist

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- - Haven't written tests yet.
- [X] New and existing unit tests pass locally with my changes
- [X] There are no merge conflicts

### CODE REVIEW

File: api/users/userModel.js [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+const removeUser = async (id) => await db('users').where({ id }).del();
+
+const findOrCreateUser = async (userObj) => {
+  const foundUser = await findUserByID(userObj.id).then((user) => user);
+  if (foundUser) {
+    return foundUser;
+  } else {
+    return await createUser(userObj).then((newUser) => {
+      return newUser ? newUser[0] : newUser;
```
Comment (antonyk) [inline api/users/userModel.js]: Logic here looks a bit confusing. Have you tested it?
Also, why are we returning newUser if newUser is false/undefined? Maybe add some error handling on failure to create?

File: api/users/userModel.js [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+      'u.email',
+      'u.phone',
+      'u.created_at'
+    );
+
+const findUserByID = async (id) =>
+  await db('users as u')
+    .where({ id })
+    .select(
+      'u.id as user_id',
```
Comment (antonyk) [inline api/users/userModel.js]: This looks like a bug: you are using "id" in the where clause but renaming to "user_id" in the select. I would keep it consistent.

File: package.json [hunk 3]
```diff
@@ -9,7 +9,8 @@
     "watch:dev": "nodemon",
     "lint": "npx eslint .",
     "lint:fix": "npx eslint --fix .",
-    "format": "npx prettier --write \"**/*.+(js|jsx|json|yml|yaml|css|md)\"",
+    "format:win": "npx prettier --write \"**/*.{js,jsx,json,yml,yaml,css,md}\"",
```
Comment (antonyk) [inline package.json:12]: nice 👍

File: api/users/userModel.js (hunk 1 above)
Comment (xanderbesse) [reply api/users/userModel.js]: The structure of the code is all provided from the scaffold. I just changed everything from the old Profiles to Users. From what I understand of the code - it's going to return the user being held at `newUser[0]` or just an empty object.

File: package.json (hunk 3 above)
Comment (Carrillo9Ana) [reply package.json:12]: that is interesting, I'm glad it's fixed now

**Review by DrakeAlia — DISMISSED** [review]
Looks good, glad the format is working

**Review by DrakeAlia — APPROVED** [review]

**Review by antonyk — APPROVED** [review]
great

**Review by Gregory-Hawman — APPROVED** [review]

**Pull request merged**